### DISCUSSION TITLE
Add "rechargeable"/"not_rechargeable" attribute to DEVICE_CLASS_BATTERY

### DISCUSSION THREAD

**Discussion by chemelli74** [home-assistant/architecture#648] [Entity Models]
Devices that are powered by battery can have a built-in rechargeable battery or not.

Having an attribute "`rechargeable`" (and corresponding "`not_rechargeable`") would allow making different automations to handle devices with those kind of batteries.

Currently only alternative is to base actions on a fixed list of `entity_id `.

**Comment by Jc2k** [comment]
We used to have a recharging/not recharging attribute on the battery, but this was pulled out into its own binary sensor. So it feels a bit weird to put this new attribute on the battery sensor.

To keep it consistent with the new world of many entities joined to a single device would this be better as it's own binary sensor, tagged with one of the new entity types (diagnostic?).

**Reply by chemelli74** [reply]
But if it will be a new sensor, there is no way to make an automation on a specific kind of devices, without relying an a name filter.

**Reply by Jc2k** [reply]
I thought there was a way to do it with "Conditions" in device automations. Shame if not.

Either way I still think it's weird to break out some battery attributes and not others.

**Comment by frenck** [comment]
This architecture issue is old, stale, and possibly obsolete. Things changed a lot over the years. Additionally, we have been moving to discussions for these architectural discussions.

For that reason, I'm going to close this issue.

../Frenck
